src/kubernetes_cluster/spec/channel.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::pervasive::prelude::*;
+use builtin::*;
+use builtin_macros::*;
+
+verus! {
+
+/// ChannelManager allocates new channel id for each request sent by the controller.
+/// The response will carry the same channel id.
+
+// TODO: the concept of channel should be modeled inside the network state machine
+pub struct ChannelManager {
+    pub cur_chan_id: nat,
+}
+
+impl ChannelManager {
+    pub open spec fn init() -> Self {
+        ChannelManager {
+            cur_chan_id: 0,
```

Comment:
Wouldn't we need to make sure there is only one global channel manager?

Reply:
Right now we have one ChannelManager per each host state machine. Do you think it is better to have one single ChannelManager at the top-level state machine?

Reply:
We need. We had some safety reasoning here: https://github.com/vmware-research/verifiable-controllers/blob/main/src/kubernetes_cluster/proof/controller_runtime_safety.rs#L32

The ChannelManager here just allocates a chan id whenever a request needs to be sent. It does not really split the network into channels. We still rely on the host state machine to correctly call `allocate`.

I thought about making the network state machine "channeled", but it seems hard and will complicate the spec a lot.